proposals/lip-0052.md

```diff
@@ -492,378 +458,817 @@ crossChainTransferParams = {
+    "type": "object",
+    "required": ["address", "nftID", "result"],
+    "properties": {
+        "address": {
+            "dataType": "bytes",
+            "length": ADDRESS_LENGTH,
+            "fieldNumber": 1
+        },
+        "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 2
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        }
+    }
+}
+```
+
+#### deleteUserEntry
+
+This event has `name = EVENT_NAME_DELETE_USER_ENTRY`, and is emitted when the deleteUserEntry function is called.
+
+##### Topics
+
+- `address`: the address to be deleted from the user substore.
+
+##### Data
+
+```java
+deleteUserEntryEventDataSchema = {
+    "type": "object",
+    "required": ["address", "nftID", "result"],
+    "properties": {
+        "address": {
+            "dataType": "bytes",
+            "length": ADDRESS_LENGTH,
+            "fieldNumber": 1
+        },
+            "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 2
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        }
+    }
+}
+```
+
+#### create
+
+This event has `name = EVENT_NAME_CREATE`, and is emitted when the create function is called.
+
+##### Topics
+
+##### Data
+
+```java
+createEventDataSchema = {
+    "type": "object",
+    "required": ["address", "collection", "attributes", "result"],
+    "properties": {
+        "address": {
+            "dataType": "bytes",
+            "length": ADDRESS_LENGTH,
+            "fieldNumber": 1
+        },
+        "collection": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        },
+        "attributes": {
+            "dataType": "bytes",
+            "maxLength": MAX_BYTE_LENGTH_ATTRIBUTES,
+            "fieldNumber": 3
+        },
+        "result": {
+            "dataType": "uint32"
+            "fieldNumber": 4
+        }
+    }
+}
+```
+
+#### destroy
+
+This event has `name = EVENT_NAME_DESTROY`, and is emitted when the destroy function is called.
+
+##### Topics
+
+##### Data
+
+```java
+destroyEventDataSchema = {
+    "type": "object",
+    "required": ["nftID", "result"],
+    "properties": {
+        "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 1
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
+```
+
+#### initializeCollection
+
+This event has `name = EVENT_NAME_INITIALIZE_COLLECTION`, and is emitted when the initializeCollection function is called.
+
+##### Topics
+
+##### Data
+
+```java
+initializeCollectionEventDataSchema = {
+    "type": "object",
+    "required": [],
+    "properties": {
+    }
+}
+```
+
+#### lock
+
+This event has `name = EVENT_NAME_LOCK`, and is emitted when the lock function is called.
+
+##### Topics
```

Comment:
I think, no topics here is fine. The same for `unlock`, `setAttributes` and `recover` events.